### ISSUE TITLE
Focus should be a type

### ISSUE THREAD

**Issue by samreid** [phetsims/joist#434]
TFocus is currently associated with Object.  We should create a new type called Focus in Focus.js and use that for type checking TFocus.  @jessegreenberg and @zepumph can you please coordinate on this?

As far as I can tell, the only place this value is set is from AccessiblePeer.js

```js
      // @private - listener for the focus event, to be disposed
      var focusEventListener = function( event ) {
        if ( event.target === self.domElement ) {
          scenery.Display.focus = {
            display: accessibleInstance.display,
            trail: accessibleInstance.trail
          };
        }
      };
```

Perhaps Focus should have this signature `Focus(display,trail)`

**Comment by jessegreenberg** [comment]
@zepumph I have added a new type for Focus called Focus.js in scenery. I believe it can be used for type checking TFocus, but not sure if there are any other steps.  Removing my assignment but let me know if I can be of help.

**Comment by samreid** [comment]
I added type checks, @zepumph can you please review?

**Comment by zepumph** [comment]
Looks good thanks
